ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1452,7 +1452,13 @@ extension SignalType {
 					var state = state
 					state.pendingValue = value
 
-					let proposedScheduleDate = state.previousDate?.dateByAddingTimeInterval(interval) ?? scheduler.currentDate
+					let proposedScheduleDate: NSDate
+					if let previousDate = state.previousDate where previousDate.compare(scheduler.currentDate) == .OrderedAscending {
```

Comment:
Sorry for the late update on the UT. I was stuck in some personal affairs and still working on that.
Let me give more detail on the situation you mentioned. `previousDate` could be after `scheduler.currentDate` if the device time is changed back during `throttle` . For example, let 's set `throttle` duration to **60 seconds**.  

| Standard Date  | Device Date | Event | 
|---|---|---|
|**00:00:00 AM**|**00:05:00 AM**|Value 0 is sent, and the `proposedScheduleDate` is **00:06:00 AM**, `previousDate` is **00:05:00 AM** |
|**00:00:30 AM**|**00:05:30 AM**|user finds the device date is 5 minutes ahead of standard date, so change device date back to **00:00:30 AM**, `proposedScheduleDate` is still **00:06:00 AM**|
|**00:00:45 AM**|**00:00:45 AM**|Value 1 is sent, and `previousDate` is **00:05:00 AM**, `scheduler.currentDate` is **00:00:45 AM**. So based on old code, `proposedScheduleDate` is **00:06:00 AM**. Thus, there will be a **6 minutes** delay (not **1 minute**) to send Next event.   | 

And the root cause of this problem is the change of device date.